docs/developer/architecture/configurations.rst

```diff
@@ -73,3 +73,205 @@ As a MetalK8s expert, I can use ``kubectl`` command(s) in order to edit all
+  - Template service pods and deployments to consume configuration data
+    directly from the above deployed ConfigMaps
+
+This approach works because in a MetalK8s cluster, cluster and service
+configurations do not influence how the cluster behaves but rather the
+BootstrapConfiguration dictates the cluster behaviour. Also the creation of
+cluster and service configurations takes place before the respective consumers
+(MetalK8s addons) are deployed hence configuration data is always ready for
+consumers and not the other way round.
+
+
+**Using Salt states**
+
+Once a ConfigMap is updated by the user (say a user changes the number of
+replicas for Prometheus deployments to a new value), then perform the
+following actions:
+
+  - Apply a Salt states that reads the ConfigMap object, validates the schema
+    and checks the new values passed.
+
+Storage format
+~~~~~~~~~~~~~~
+
+A YAML(K8s-like) format was chosen to represent the data field instead of a
+flat key-value structure for the following reasons:
+
+ - YAML formatted configurations are easy to write and understand hence it will
+   be simpler for users to edit configurations.
+ - The YAML format benefits from bearing a schema version, which can be checked
+   before inferring the underlying service configuration if need arises.
+ - The YAML format utilizes white spaces to define variable hierarchies and as
+   such it is easier to manage with code mean while the flat key-value format
+   requires parsing of flat keys into hierarchical structures which can be
+   tedious.
+
+A sample ConfigMap can be defined with the following fields.
+
+An example of such a ConfigMap:
+
+.. code-block:: yaml
+
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      namespace: <namespace>
+      name: <config-name>
+    data:
+      config.yaml: |-
+        apiVersion: <object-version>
+        kind: <kind>
+        spec:
+          <key>: <values>
+
+**Use case 1:**
+
+Configure and store the number of replicas for service specific Deployments
+found in the `metalk8s-monitoring` namespace using the ConfigMap format.
+
+.. code-block:: yaml
+
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      namespace: metalk8s-monitoring
+      name: metalk8s-grafana-config
+    data:
+      config.yaml: |-
+        apiVersion: metalk8s.scality.com/v1alpha1
+        kind: GrafanaConfig
+        spec:
+          deployment:
+            replicas: 2
+
+Non-goals
+~~~~~~~~~
+
+This section contains requirements stated above which the current design choice
+does not cater for and will be addressed later:
+
+- Persisting newly added Grafana dashboards or new Grafana datasources
+  especially for modifications added via the Grafana UI cannot be stored in
+  ConfigMaps and hence will be catered for later.
+
+- As stated in the requirements, adding and editing Prometheus alert rules
+  is also not covered by the chosen design choice and will need to be addressed
+  differently.
```

Comment:
Not clear enough why we don't cover Prometheus rules (and not only alerting rules are affected, btw). Explain that even if we **could** use ConfigMaps for said rules, we **prefer** relying on the Prometheus Operator (which we deploy) and its CRDs (especially, the `PrometheusRule` one).

Reply:
As earlier indicated by Nicolas, these are cases that our design does not cater for and it is not required state the solution we want to use.